src/ol/coordinate.js

```diff
@@ -410,17 +410,22 @@ export function toStringXY(coordinate, opt_fractionDigits) {
  * @return {Coordinate} The coordinate within the real world extent.
  */
 export function wrapX(coordinate, projection) {
+  const worldsAway = getWorldsAway(coordinate, projection);
+  if (worldsAway) {
+    coordinate[0] -= worldsAway * getWidth(projection.getExtent());
+  }
+  return coordinate;
+}
+
+export function getWorldsAway(coordinate, projection) {
```

Comment:
If this were something like `getWorldOffset` and returned `worldsAway * extentWidth`, we could avoid calculating the extent width again in the places it is called.

Reply:
I see now that it is also used on the transformed coordinate.  It could take an extent width instead of a projection, but perhaps not as nice.

Reply:
I solved this now by optionally providing an already calculated world width.